README.md

```diff
@@ -1,85 +1,142 @@
+
+The f18 CMakeList.txt file uses
+the variable `LLVM_DIR` to find the installed components.
+
+To get the correct LLVM libraries included in your f18 build,
+define LLVM_DIR on the cmake command line.
+```
+LLVM=<LLVM_INSTALLATION_DIR>/lib/cmake/llvm cmake -DLLVM_DIR=$LLVM ...
+```
+where `LLVM_INSTALLATION_DIR` is
+the top-level directory
+where llvm is installed.
+
+### Building f18 with GCC
 
 By default,
 cmake will search for g++ on your PATH.
-The g++ version must be 7.2 or greater in order to build f18.
+The g++ version must be one of the supported versions
+in order to build f18.
 
-Or, if you export CXX,
+Or,
 cmake will use the variable CXX to find the C++ compiler.
 CXX should include the full path to the compiler
 or a name that will be found on your PATH,
 e.g. g++-7.2, assuming g++-7.2 is on your PATH.
 ```
 export CXX=g++-7.2
 ```
+or
+```
+CXX=/opt/gcc-7.2/bin/g++-7.2 cmake ...
+```
+There's a third option!
+The CMakeList.txt file uses the variable GCC
+as the path to the bin directory containing the C++ compiler.
 
-Or, you can reference the GCC installation directory directly.
-The CMakeList.txt file
-uses the variable GCC
-as the path to the bin directory
-containing the C++ compiler.
 GCC can be defined on the cmake command line
 where `<GCC_DIRECTORY>` is the path to a GCC installation with bin, lib, etc:
 ```
-cmake -DGCC=<GCC_DIRECTORY>
+cmake -DGCC=<GCC_DIRECTORY> ...
 ```
 
-To use f18 after it is built,
-the environment variables PATH and LD_LIBRARY_PATH
-must be set to use GCC and its associated libraries.
+### Building f18 with clang
 
-### LLVM and Clang dependency
+To build f18 with clang,
+cmake needs to know how to find clang++
+and the GCC library and tools that were used to build clang++.
 
-F18 uses components from version 6.0 of LLVM and clang
-(even when f18 is not compiled with clang).
+The CMakeList.txt file expects either CXX or BUILD_WITH_CLANG to be set.
 
-The instructions to build LLVM and clang can be found at
-https://clang.llvm.org/get_started.html.
+CXX should include the full path to clang++
+or clang++ should be found on your PATH.
+```
+export CXX=clang++
+```
+BUILD_WITH_CLANG can be defined on the cmake command line
+where `<CLANG_DIRECTORY>`
+is the path to a clang installation with bin, lib, etc:
+```
+cmake -DBUILD_WITH_CLANG=<CLANG_DIRECTORY>
+```
 
-The f18 CMakeList.txt file uses
-the environment variable `Clang_DIR` to find the installed components.
+To use f18 after it is built,
+the environment variables PATH and LD_LIBRARY_PATH
+must be set to use GCC and its associated libraries.
 
-To get the correct LLVM and clang libraries included in your f18 build,
-set the environment variable
-`Clang_DIR`
-to the `lib/cmake/clang` directory in the clang install directory.
+Or GCC can be defined on the f18 cmake command line
+where `<GCC_DIRECTORY>` is the path to a GCC installation with bin, lib, etc:
+```
+cmake -DGCC=<GCC_DIRECTORY> ...
+```
```

Comment:
It looks like this paragraph isn't supposed to be here. It's not an alternative to the previous sentence.

Reply:
Reordered so it makes sense.